### PR TITLE
add nullok support to challenge-response mode

### PR DESCRIPTION
**Pull request by Larhard**
Hi, I wanted to use nullok flag with challenge-response mode so I added the support. I hope it will be helpful.

It solves issue "Cannot login with nullok using challenge-response? #119"

### CODE REVIEW

File: util.c [hunk 1]
```diff
@@ -282,6 +283,64 @@ int challenge_response(YK_KEY *yk, int slot,
...
 
 int
+check_user_challenge_file(const char *chalresp_path, const struct passwd *user, FILE *debug_file)
+{
+  /*
+   * This function will look for users challenge files.
+   *
+   * Returns one of AUTH_FOUND, AUTH_NOT_FOUND, AUTH_ERROR
+   */
+  int len;
```
Comment (klali) [inline util.c]: len should probably be a size_t since it's assigned result of strlen(), passed to malloc() and passed to snprintf()

File: util.c [hunk 2]
```diff
@@ -282,6 +283,64 @@ int challenge_response(YK_KEY *yk, int slot,
 }
 
 int
+check_user_challenge_file(const char *chalresp_path, const struct passwd *user, FILE *debug_file)
```
Comment (klali) [inline util.c]: debug_file isn't used in the function, consider dropping it?

File: util.c [hunk 3]
```diff
@@ -282,6 +283,64 @@ int challenge_response(YK_KEY *yk, int slot,
...
+
+  len = strlen(userfile) + 2 + 1;
+  if ((userfile_pattern = malloc(len)) == NULL) {
+    goto out;
+  }
+  snprintf(userfile_pattern, len, "%s-*", userfile);
+
+  r = glob(userfile_pattern, 0, NULL, &userfile_glob);
+  globfree(&userfile_glob);
+  switch (r) {
```
Comment (klali) [inline util.c]: I'd prefer for all cases in the switch to look the same, setting ret+break, this decreases risks of messing this up in the future.

**Comment by Larhard** [comment]
In a free moment I will fix this, but for now I didn't have any.

File: util.c (hunk 2 above)
Comment (Larhard) [inline util.c]: I've added some error messages instead

File: util.c (hunk 3 above)
Comment (Larhard) [inline util.c]: Comment shoud explain my idea. I would keep it that way (moved break to the top and then ret+goto out)

**Comment by klali** [comment]
Ok. Merging this, thank you!

**Pull request merged**